rfcs/expected_fail_message_tag_for_subtests.md

```diff
@@ -0,0 +1,71 @@
+# RFC 165: Support expected-fail-message in test metadata for subtests
+
+## Summary
+
+Add "expected-fail-message" in test metadata so that we can track how
+a subtest fails.
+
+## Details
+
+In Chromium developers are using baselines to track expected output message
+for testharness tests. While the output message for a pass subtest is
+generally not interesting, the output message for a failed subtest tracks
+how a subtest fails, and a change in how a test fails could be unintended
+and suggestive of a problem. Chromium developers want to keep such capability
+if we want to use Wptrunner to run WPTs in Chromium.
+
+The suggestion is to support "expected-fail-message" tag for subtests. When
+present, it takes effect on expected FAIL or expected PRECONDITION_FAILED
+results. The test runner should check if the actual output message matches
+with the "expected-fail-message". When a mismatch is found, the test runner
+should turn the results to unexpected FAIL or unexpected PRECONDITION_FAILED
+respectively, otherwise the results continue to be the expected one.
+
+Wdspec tests can also have subtests. "expected-fail-message" should work the
+same way for an expected FAIL Wdspec subtest. Other than that, "expected-fail-message"
+should not have impact on any other scenarios.
+
+One example metadata file will look like below:
+```
+[event.html]
+  expected: OK
+  [Changing selection from handler: input event]
+    expected: FAIL
+
+  [Command createLink, value "": input event]
+    expected: [FAIL, PRECONDITION_FAILED, PASS]
+    expected-fail-message: "number of input events fired expectei 1 but got 0"
+```
+
+### Implementations
```

Comment:
Will you also update the manifest update code to allow updating the `expected-fail-message` metadata? If so this needs to be an optional flag so that only people who are using this code get the additional keys.

Reply:
Added
```
How to add "expected-fail-message" to test metadata is out of scope of this RFC.
Usually this tag can be added manually by developers after carefully review the
test results and implementations.
```

I don't want to make this process automatically, in case it could be abused.

Reply:
OK, but I don't understand how that will work with your sync. If you import a bunch of tests which have changed their failure messages will you have to manually update any `expected-fail-message` data?

Reply:
We will continue use baselines, and when running test, recreate test metadata from the baselines.

So for us, we will use `expected-fail-message` if there is a fail message from the baseline.